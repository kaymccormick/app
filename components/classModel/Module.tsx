import React from 'react';

interface ModuleProps {
    name: string;
    id: number;
}
export default class Module extends React.Component<ModuleProps> {

    render() {
        return <div className="module"><div className="module__name">{this.props.name}</div>
        </div>;
    }
}
