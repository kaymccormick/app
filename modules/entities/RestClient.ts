import axios from 'axios';
import GetEntitiesResponse from './rest/response/GetEntitiesResponse';
export interface Args {
    baseUri: string;
}

export default class RestClient {
    public baseUri: string;

    public constructor(args: Args) {
        this.baseUri = args.baseUri;
    }
    public getEntities(): Promise<GetEntitiesResponse> {
        return axios.get(`${this.baseUri}/entity`).then(response => {
            if(response.status >= 400) {
                return new Error();
            }
            return response.data;
        }).then(data => {
            const r = new GetEntitiesResponse(data.success, data.result);
            return r;
        });
    }
}
