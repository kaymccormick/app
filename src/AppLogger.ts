export class AppLogger {
debug(message: string, meta?: any) {
let meta2;
if(meta !== undefined) {
  meta2 = Object.assign({}, meta);
delete meta2.msg;
  } else {
  meta2 = {};
  }
let myMsg = message;
if(meta && meta.msg) {
  myMsg += meta.msg;
}
console.log(myMsg);
}
}
