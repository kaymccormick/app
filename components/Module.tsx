import React from 'react';
import { Registry, Module } from 'classModel';

export default (props: {key: string; module: Module}) => <div>
    <div>hi {props.key}</div>
</div>;
