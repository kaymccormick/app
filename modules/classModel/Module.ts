import {ApplicationModule, ApplicationModuleArgs} from '../../src/ApplicationModule';
import {Configuration} from '../../src/Configuration';
import {Pojo} from 'classModel';
import reducer from './reducers';
import actions from './actions';
import { WebApplication } from '../../src/WebApplication';
import {ModuleState, EntitiesType} from './types';

import {Map,List,Set} from 'immutable';

import RestClient from '@heptet/rest-client'
/*export type Args = {
  restClient:RestClient,
}*/

export class Module extends ApplicationModule<ModuleState> {
    public restClient?: RestClient;
    public name: string = 'classModel';
    public key: string = 'classModel';

    public constructor(args: ApplicationModuleArgs) {
        super(args);
        this.actions = actions(args.restClient);
        this.restClient = args.restClient;
    }

    public getInitialState(): ModuleState {
        return {
            entities: Map<string, Map<number, Pojo.BasePojo>>(),
        }
    }
    

    public getReducers(): any {
        return reducer({restClient:this.restClient!});

    }
    public setup(app: WebApplication, config: Configuration) {
    }

    public getMainComponent(): Promise<any> {
        return import('./components/MainComponent');
    }
}


