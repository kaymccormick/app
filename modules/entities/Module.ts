import {ApplicationModule} from '../../src/ApplicationModule';
import {Configuration} from '../../src/Configuration';
import reducer from './reducers';
import * as actions from './actions';
import {RestClient} from './RestClient';
import { EntitiesViewContainer } from './components/containers/EntitiesViewContainer';
import { WebApplication } from '../../src/WebApplication';
import uuidv4 from 'uuid/v4';

export class Module extends ApplicationModule {
    public restClient?: RestClient;
    public name: string = 'entities';
    public key: string = 'entities';
    public id: string;

    public constructor() {
        super();
        this.id = uuidv4();
        this.actions = actions;
//        console.log(`${this.id} setting rest client`);
        this.restClient = new RestClient({ baseUri: '/cme'});
//        console.log(`${this.id} ${this.restClient}`);
    }

    public getInitialState(): any {
        return {
            entities: [],
        }
    }

    public getReducers(): any {
        if(!this.restClient) {
            throw new Error(`${this.id} No rest client`);
        }
        const r = reducer({module: this,restClient:this.restClient});
        console.log(r);
        return r;

    }
    public setup(app: WebApplication, config: Configuration) {
        //const restClient = config.getResource('restClient');
        //this.restClient = restClient;
    }

    public getMainComponent(): any {
        return EntitiesViewContainer;
    }

}

