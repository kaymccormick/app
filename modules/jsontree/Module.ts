import uuidv4 from 'uuid/v4';
import {List,Map} from 'immutable';
import {ApplicationModule, ApplicationModuleArgs} from '../../src/ApplicationModule';
import {Configuration} from '../../src/Configuration';
import reducer from './reducers';
import * as actions from './actions';
import { WebApplication } from '../../src/WebApplication';
import {ModuleState} from './types';

export class Module extends ApplicationModule<ModuleState> {
    public name: string = 'jsontree';
    public key: string = 'jsontree';
    public id: string;

    public constructor(args: ApplicationModuleArgs) {
        super(args);
        this.id = uuidv4();
        this.actions = actions;
    }

    public getInitialState(): ModuleState {
        return {
        }
    }

    public getReducers(): any {
        return reducer({module: this});
    }
    public setup(app: WebApplication, config: Configuration) {
    }
    public getMainComponent(): Promise<any> {
        return import('./components/MainComponent');
    }
}

        