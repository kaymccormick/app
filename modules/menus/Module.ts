import uuidv4 from 'uuid/v4';
import {List,Map,Set} from 'immutable';
import {ApplicationModule,ApplicationModuleArgs} from '../../src/ApplicationModule';
import {Configuration} from '../../src/Configuration';
import reducer from './reducers';
import * as actions from './actions';
import { WebApplication } from '../../src/WebApplication';
import {MenuItemPojo, ModuleState} from './types';

export class Module extends ApplicationModule<ModuleState> {
    public name: string = 'menus';
    public key: string = 'menus';
    public id: string;

    public constructor(args: ApplicationModuleArgs) {
        super(args);
        this.id = uuidv4();
        this.actions = actions;
    }

    public getInitialState(): ModuleState {
        return {
            menuItems: Map<string, MenuItemPojo>({'': { key: '', title: 'root',
                subItems: Set<string>(),
            }}),
        }
    }

    public getReducers(): any {
        return reducer({module: this});
    }
    public setup(app: WebApplication, config: Configuration) {
    }
    public getMainComponent(): Promise<any> {
        return import('./components/MainComponent');
    }
}

