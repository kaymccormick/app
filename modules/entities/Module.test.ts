import { Module } from './Module';
import { WebApplication } from '../../src/WebApplication';
import { Configuration } from '../../src/Configuration';
import{ AppLogger} from '../../src/AppLogger';
import RestClient from '@heptet/rest-client/src/RestClient';
jest.mock('@heptet/rest-client/src/RestClient');
import { createRestClient,createAppLogger } from '../../src/testUtils';

jest.mock('../../src/WebApplication');
jest.mock('../../src/Configuration');

let appLogger: AppLogger;
beforeEach(() => {
    appLogger = createAppLogger();
    // @ts-ignore
    WebApplication.mockClear();
    // @ts-ignore
    Configuration.mockClear();
    // @ts-ignore
    RestClient.mockClear();
});

test('Module.constructor', () => {
    const restClient = createRestClient();
    const m = new Module({logger: appLogger, restClient});
    expect(m).toBeDefined();
    expect(m.id).toBeDefined();
    expect(m.actions).toBeDefined();
    expect(m.restClient).toBeDefined();
});

test('Module.getInitialState', () => {
    const restClient = createRestClient();
    const m = new Module({logger: appLogger, restClient});
    const state = m.getInitialState();
    expect(state).toBeDefined();
    expect(state.entities).toStrictEqual([]);
});
test('Module.getReducers', () => {
    const restClient = createRestClient();
    const m = new Module({logger: appLogger, restClient});
    const reducers = m.getReducers();
    expect(reducers).toBeDefined();
    expect(Object.keys(reducers)).toMatchSnapshot();
});
