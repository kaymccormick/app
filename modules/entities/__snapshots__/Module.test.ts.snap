// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module.getReducers 1`] = `Array []`;
